controllers/update_pods.go

```diff
@@ -166,6 +175,22 @@ func shouldRequeueDueToTerminatingPod(pod *corev1.Pod, cluster *fdbv1beta2.Found
 		!cluster.ProcessGroupIsBeingRemoved(processGroupID)
 }
 
+// Because the internal.fault_tolerance relies in fdbcli status only we need to make sure it is
+// synced with the cluster status to prevent race conditions.
+func shouldRequeueDueToClusterStatusDrift(adminClient fdbadminclient.AdminClient, podMap map[string]*corev1.Pod) (bool, error) {
+	status, err := adminClient.GetStatus()
+	if err != nil {
+		return false, err
+	}
+	//Check that all processes in fdbcli status have a matching pod.
+	for _, p := range status.Cluster.Processes {
+		if _, ok := podMap[p.Locality[fdbv1beta2.FDBLocalityInstanceIDKey]]; !ok {
```

Comment:
I think this will probably not solve the actual issue, since a Pod that is in `Pending` state would be ignored here and Pods that are stuck in Pending (for whatever reason) would stay there and the operator would proceed deleting Pods once the data is redistributed. What we probably want/need is to implement a logic like `maxUnavailable ` from the Deployment (See: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#max-unavailable), that would ensure that the operator stops removing Pods after x Pods went into an unavailable state (in our case `MissingProcesses` or `PodPending`). I open an issue for this.